connman/vpn/vpn-provider.c

```diff
@@ -3289,6 +3301,36 @@ const char *vpn_provider_get_path(struct vpn_provider *provider)
 unsigned int vpn_provider_get_authentication_errors(
 						struct vpn_provider *provider)
 {
+	const char *val;
+	unsigned int limit;
+
+	val = vpn_provider_get_string(provider, "AuthErrorLimit");
+	if (val)
+		limit = g_strtoull(val, NULL, 10);
+	else
+		limit = ALLOW_AUTH_ERRORS_DEFAULT;
+
+	DBG("%p errors %d limit %d", provider, provider->auth_error_counter,
+									limit);
+
+	/*
+	 * This crude heuristic is for VPN providers who require that the VPN
+	 * in use does complete clean shutdown process, which is not possible
+	 * if the underlying network used as transport dies before that and
+	 * there is no route to server anymore. In such case it really seems
+	 * that authentication error is reported back by, at least OpenVPN
+	 * binary but allowing it to try again if we had previously succesful
+	 * connection with the same provider, which will not result in
+	 * potentially unnecessary credential resets.
+	 */
+	if (provider->conn_error_counter == 0 &&
+				provider->previous_connect_status == true &&
+				provider->auth_error_counter < limit) {
```

Comment:
I'm guessing this should be <=

Reply:
Yeah, true. Changed with bc93dcd0c1ea3433cf623d7fcec141559eac5fd5